lambdas/gro-convert-to-json/src/main/java/uk/gov/di/data/lep/GroConvertToJson.java

```diff
@@ -146,4 +150,18 @@ private void generateAndAddListOfAuditDataToQueue(List<GroJsonRecordWithHeaders>
             addAuditDataToQueue(auditData);
         }
     }
+
+    @Tracing
+    private String uploadIndividualRegistrations(String bucket, List<GroJsonRecordWithHeaders> deathRegistrations) throws JsonProcessingException {
+        var jsonKeys = new ArrayList<RecordLocation>();
+        for (var registration : deathRegistrations) {
+            var registrationJsonKey = registration.correlationID() + ".json";
+            jsonKeys.add(new RecordLocation(bucket, registrationJsonKey));
+            awsService.putInBucket(bucket, registrationJsonKey, objectMapper.writeValueAsString(registration));
+        }
+        var jsonKey = UUID.randomUUID() + ".json";
```

Comment:
I think I'd prefer these to be deterministic. If the function times out, or there's a hiccup in S3 or something else that causes this function to error, we'd be left with a load of orphan objects. An S3 put is an upsert, so if the filenames were consistent then a retry would succeed (assuming the original issue was resolved) and the files would be cleaned up properly